File: src/WebApp/CommerceCore.Web.AdminSite/src/shared/constants/category.ts
type CategoryLabelKey =
  | "id"
  | "name"
  | "description"
  | "createdDate"
  | "updatedDate"
  | "parentCategoryId"
  | "parentCategory";

export const categoryLabels: Record<CategoryLabelKey, string> = {
  id: "Id",
  name: "Name",
  description: "Description",
  createdDate: "Created Date",
  updatedDate: "Updated Date",
  parentCategoryId: "Parent Category Id",
  parentCategory: "Parent Category",
};

File: src/ResourceServer/CommerceCore.Domain/Entities/Category.cs
namespace CommerceCore.Domain.Entities;

public class Category
{
    public Guid Id { get; set; }
    public required string Name { get; init; }
    public required string Description { get; init; }
    public Guid? ParentCategoryId { get; init; }
    public DateTime CreatedDate { get; init; }
    public DateTime UpdatedDate { get; init; }

    public virtual Category? ParentCategory { get; init; }
    public virtual AppImage? Image { get; init; }
}
